### PR TITLE
Add deekseek distill golden logits generation.

### PR DESCRIPTION
**Pull request by wang2yn84**
# Description

Deekseek distilled llama model is structurally the same as llama 3.1 8B and 3.3 70B model. Therefore we reuse the llama3.1 golden logit generation script.

# Tests

Manually generated the checkpoint.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ ] I have performed a self-review of my code.
- [ ] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ ] I have run end-to-end tests tests and provided workload links above if applicable.
- [ ] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/scratch_code/golden_llama3_1_export.py [hunk 1]
```diff
@@ -13,8 +13,16 @@
...
 
-Usage: python3 -m MaxText.scratch_code.golden_llama3_1_export --model-id meta-llama/Meta-Llama-3-70B \
-  --output-path llama3-70b/golden_logits/golden_data_llama3-70b.jsonl
+Usage:
+
+python3 -m MaxText.scratch_code.golden_llama3_1_export --model-id meta-llama/Meta-Llama-3-70B \
+  --output-path maxtext-test-assets/golden_logits/golden_data_llama3-70b.jsonl
+
+python3 -m MaxText.scratch_code.golden_llama3_1_export --model-id deepseek-ai/DeepSeek-R1-Distill-Llama-70B \
+  --output-path maxtext-test-assets/golden_data_deepseek-r1-distill-llama3-70b.jsonl
```
Comment (shralex) [inline MaxText/scratch_code/golden_llama3_1_export.py:22]: Notice that your code now uses output-path as the blob id So I think it needs to be different in the 3 examples.

Comment (wang2yn84) [reply MaxText/scratch_code/golden_llama3_1_export.py]: That's true. I updated the script to make bucket name a variable. Ptal.

**Review by shralex — APPROVED** [review]
Thanks Lance! If these generated files are used somewhere in MaxText, e.g., end-to-end tests, could you please add a comment that points to this file ? And perhaps here end_to_end/tpu/deepseek/Run_DeepSeek.md

**Review by RissyRan — COMMENTED** [review]
If you are making it in general, is that possible to reuse this one? https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/scratch_code/generate_hf_golden_logits.py

File: MaxText/scratch_code/golden_llama3_1_export.py [hunk 2]
```diff
@@ -13,8 +13,16 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 
-Usage: python3 -m MaxText.scratch_code.golden_llama3_1_export --model-id meta-llama/Meta-Llama-3-70B \
-  --output-path llama3-70b/golden_logits/golden_data_llama3-70b.jsonl
+Usage:
+
+python3 -m MaxText.scratch_code.golden_llama3_1_export --model-id meta-llama/Meta-Llama-3-70B \
```
[the remark this answers is not in the document]
Comment (wang2yn84) [reply MaxText/scratch_code/golden_llama3_1_export.py:18]: Hi Alex, the naming is explanatory, wdyt?

**Comment by wang2yn84** [comment]
> If you are making it in general, is that possible to reuse this one? https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/scratch_code/generate_hf_golden_logits.py

Hi Ran, I agree someone should refactor these 2 scripts if their functionality are completely overlapped. The reason why I put deepseek distill model and llama model together is because their model structure are exactly the same. This PR only adds the missing piece (deepseek distilled models).

File: MaxText/scratch_code/golden_llama3_1_export.py [hunk 3]
```diff
@@ -13,8 +13,16 @@
...
+  --output-path golden_logits/golden_data_llama3-70b.jsonl
+
+python3 -m MaxText.scratch_code.golden_llama3_1_export --model-id deepseek-ai/DeepSeek-R1-Distill-Llama-70B \
+  --output-path golden_data_deepseek-r1-distill-llama3-70b.jsonl
+
+python3 -m MaxText.scratch_code.golden_llama3_1_export --model-id deepseek-ai/DeepSeek-R1-Distill-Llama-8B \
+  --output-path golden_data_deepseek-r1-distill-llama3-8b.jsonl
```
Comment (bvandermoon) [inline MaxText/scratch_code/golden_llama3_1_export.py:25]: Do the DeepSeek examples need a different `output_bucket` than the default `maxtext-llama`? Or are you intentionally keeping them in the same place due to distillation?

Comment (wang2yn84) [reply MaxText/scratch_code/golden_llama3_1_export.py:25]: Hi Branden, the actual output bucket for the logits are maxtext-test-assets, but I don't want to default it there and to overwrite them by accident. @shralex , are we still using maxtext-llama bucket for anything?

**Comment by RissyRan** [comment]
> > If you are making it in general, is that possible to reuse this one? https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/scratch_code/generate_hf_golden_logits.py
> 
> Hi Ran, I agree someone should refactor these 2 scripts if their functionality are completely overlapped. The reason why I put deepseek distill model and llama model together is because their model structure are exactly the same. This PR only adds the missing piece (deepseek distilled models).

Thanks Lance! Personally I feel they are the same only line diffence may be `tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-3.1-8B")` that can be passed from `model-id` ([link](https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/scratch_code/generate_hf_golden_logits.py#L79))? 

I saw you are refactoring it, so maybe it's an opportunities to reduce the duplicated scripts?

**Comment by wang2yn84** [comment]
> > > If you are making it in general, is that possible to reuse this one? https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/scratch_code/generate_hf_golden_logits.py
> > 
> > 
> > Hi Ran, I agree someone should refactor these 2 scripts if their functionality are completely overlapped. The reason why I put deepseek distill model and llama model together is because their model structure are exactly the same. This PR only adds the missing piece (deepseek distilled models).
> 
> Thanks Lance! Personally I feel they are the same only line diffence may be `tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-3.1-8B")` that can be passed from `model-id` ([link](https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/scratch_code/generate_hf_golden_logits.py#L79))?
> 
> I saw you are refactoring it, so maybe it's an opportunities to reduce the duplicated scripts?

Hi Ran, refactoring scripts is not the goal of this PR. The priority is to have a script that can reproduce the deepseek distilled model logits. Refactoring another script is not simply change it, but also need to validate all the affected models to make sure all of the model working properly. So I'd prefer to limit the scope to this single script. IMO, you should refactor generate_hf_golden_logits.py, golden_llama2-70b_export.py and golden_llama3_1_export.py in the future, but again, that's not the purpose of this PR.
